Description: Probe all devices if we've been asked for a drive name
 This allows --target=drive to work properly even without a device.map.
 .
 Depends on ubuntu_lvm_raid_probe.patch.
Author: Colin Watson <cjwatson@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/549980
Forwarded: no
Last-Update: 2010-07-05

Index: b/util/grub-probe.c
===================================================================
--- a/util/grub-probe.c
+++ b/util/grub-probe.c
@@ -142,7 +142,9 @@
     }
 
   if (stat (dev_map, &dev_map_stat) == -1 &&
-      grub_util_get_dev_abstraction (device_name) != GRUB_DEV_ABSTRACTION_NONE)
+      (print == PRINT_DRIVE ||
+       grub_util_get_dev_abstraction (device_name) !=
+	 GRUB_DEV_ABSTRACTION_NONE))
     {
       /* If we don't have a device map, then we won't yet know about the
          physical volumes underlying this device, so probe all devices.  */
